File: CS-IOS/Classes/Core/Classes/UI/CSMenuHeader.m
//
// Created by Rene Dohan on 14/03/18.
//

#import "CSMenuHeader.h"
#import "NSObject+CSExtension.h"
#import "NSMutableArray+CSExtension.h"
#import "CSMenuItem.h"
#import "CSMainController.h"

@implementation CSMenuHeader {
    CSMainController *_controller;
}

- (instancetype)construct:(CSMainController *)controller :(int)index :(NSString *)title {
    super.construct;
    _controller = controller;
    _index = index;
    _title = title;
    _items = NSMutableArray.new;
    return self;
}

- (CSMenuItem *)item:(NSString *)name {
    return [_items add:[CSMenuItem.new construct:_controller :_title]];
}

- (instancetype)clear {
    _items.removeAllObjects;
    return self;
}

@end

File: Renetik/Renetik/Classes/Core/framework/controller/Menu/CSMenuHeader.swift
//
// Created by Rene Dohan on 12/18/19.
//

import Foundation

public class CSMenuHeader: CSObject {

    private let controller: CSMainController

    var items = [CSMenuItem]()
    let index: Int
    let title: String

    public init(by parent: CSMainController, index: Int, title: String) {
        self.controller = parent
        self.index = index
        self.title = title
    }

    public func item(with title: String, type: UIBarButtonItem.SystemItem? = nil,
                     image: UIImage? = nil, action: ((CSMenuItem) -> Void)? = nil) -> CSMenuItem {
        items.add(CSMenuItem(by: controller, title: title, systemItem: type, image: image, action: action))
                .also { $0.index = items.size - 1 }
    }

    public func item(with view: UIView, action: ((CSMenuItem) -> Void)? = nil) -> CSMenuItem {
        items.add(CSMenuItem(by: controller, view: view, action: action)).also { $0.index = items.size - 1 }
    }

    public var isVisible: Bool {
        for item in items { if item.isVisible { return true } }
        return false
    }

    public func clear() -> Self {
        items.removeAll()
        return self
    }

    public var isDisplayedAsItem: Bool { title.isEmpty && items.count == 1 && !items.first!.isNoActionItem }
}
